docs/src/passing-in-data.rst

```diff
@@ -83,15 +84,75 @@ which gets the value of the environment variable **MYBUCKET**.
 Getting a Value from an SSM Store Variable
 ==========================================
 
-To get the value of an SSM parameter store variable,
-use code like the following,
-which uses the value of the SSM variable **my-awesome-value**.
+There are three ways to get the value of an SSM parameter store variable, depending on whether you want
+the latest version of a plain string, a particular version of a plain string, or a particular version
+of a secret string. It is not possible to retrieve the latest version of a secure string. To read the
+latest version of a secret, you have to read the secret from SecretsManager (see `doc:using_value_from_secrets_manager`).
+The first two are not recommended for values that are supposed to be secrets, such as passwords.
+
+To retrieve the latest value once (as a context value, see the :doc:`context` topic), and keep on using the same value
+until the context value manually refreshed, use a :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`:
+
+.. code-block:: ts
+
+    import cdk = require('@aws-cdk/cdk');
+
+    const myvalue = new cdk.SSMParameterProvider(this).getString("my-parameter-name");
+
+To read a particular version of an SSM Parameter Store plain string value at CloudFormation deployment time,
+use :py:class:`SsmParameterStoreString <@aws-cdk/cdk.SsmParameterStoreString>`:
 
 .. code-block:: ts
 
-    const myvalue = new SSMParameterProvider(this).getString("my-awesome-value");
+    import ssm = require('@aws-cdk/aws-ssm');
+
+    const parameterString = new ssm.ParameterStoreString(this, 'MyParameter', {
+        parameterName: 'my-parameter-name',
+        version: 1,
+    });
+    const myvalue = parameterString.value;
+
+To read a particular version of an SSM Parameter Store SecureString value at CloudFormation deployment time,
+use :py:class:`SsmParameterStoreSecureString <@aws-cdk/cdk.SsmParameterStoreSecureString>`:
+
+.. code-block:: ts
+
+    import ssm = require('@aws-cdk/aws-ssm');
+
+    const secureString = new ssm.ParameterStoreSecureString(this, 'MySecretParameter', {
+        parameterName: 'my-secret-parameter-name',
+        version: 1,
+    });
+    const myvalue = secureString.value;
 
-See the :doc:`context` topic for information about the :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`.
+
+.. _using_value_from_secrets_manager:
+
+Getting a Value from AWS Secrets Manager
+========================================
+
+To use values from AWS Secrets Manager in your CDK app, create an instance of :py:class:`SecretsManagerValue
+<@aws-cdk/cdk.SecretsManagerValue>`. It represents a value that is retrieved from Secrets Manager and used
+at CloudFormation deployment time.
+
+.. code-block:: ts
+
+    import secretsmanager = require('@aws-cdk/aws-secretsmanager');
+
+    const loginSecret = new secretsmanager.SecretString(stack, 'Secret', {
+      secretId: 'MyLogin'
+
+      // By default, the latest version is retrieved. It's possible to
```

Comment:
Optional - I like to use a different comment style for "commented code" and "text", e.g. use `/*` style for text. I find this makes it easier to spot opt-in code fragments.